lib/money-rails/active_record/monetizable.rb

```diff
@@ -55,6 +55,9 @@ def monetize(*fields)
                                    "Use :as option to explicitly specify the name or change the amount column postfix in the initializer."
             end
 
+            # Assigns infinite_precision based on the field's column type, if available
+            options[:infinite_precision] = self.columns_hash[subunit_name].try(:type) == :decimal
```

Comment:
Why store this in the `options` hash, just to slice it out later? Is there any code that currently accepts the `options` hash, that needs the `:infinite_precision` value? Seems like `options` is used to pass values _into_ this method.

How about:

```suggestion
            column_definition = columns_hash[subunit_name]
            money_options = {
              :infinite_precision => !!(column_definition && column_definition.type == :decimal)
            }
```

and then pass `money_options` to Money constructors.

Reply:
Updated suggestion to remove use of ActiveSupport's `try`, since its behavior is confusing (and not usually what the author intended - see `try!`).

Reply:
The options hash was reused for two reasons:
1. I initially had an option to set an attribute to `infinite_precision` directly that took precedence over the column type.  As you pointed out in our Slack convo the backing column could be a string, in which case we might want to allow setting the option directly after all.
2. To avoid changing the arity of the `read_monetized` and `write_monetized` methods... however, they do appear to be effectively private to the class, so changing the arity may not be an issue.

Reply:
1. Yes, I think you probably should allow overriding, and only infer the value from `columns_hash` if it hasn't been specified.

2. I see now. I didn't realize use of the `options` hash spanned across multiple methods.

Reply:
Implemented per our Slack convo with backwards compatibility and an `infer_precision` boolean option.